aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -42,63 +40,105 @@ module aptos_framework::fungible_asset {
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The wallet object that holds fungible assets of a specific type associated with an account.
     struct FungibleAssetWallet has key {
         /// The address of the base metadata object.
         metadata: Object<FungibleAssetMetadata>,
         /// The balance of the fungible metadata.
         balance: u64,
-        /// Fungible Assets transferring is a common operation, this allows for disabling and enabling
-        /// transfers bypassing the use of a TransferRef.
+        /// Fungible Assets transferring is a common operation, this allows for freezing/unfreezing accounts.
         allow_ungated_transfer: bool,
-        /// The delete_ref of this object, used for cleanup.
-        delete_ref: DeleteRef
     }
 
-    /// The transferable version of fungible metadata.
-    /// Note: it does not have `store` ability, only used in hot potato pattern.
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct FungibleAssetWalletEvents has key {
+        deposit_events: event::EventHandle<DepositEvent>,
+        withdraw_events: event::EventHandle<WithdrawEvent>,
+        set_ungated_transfer_events: event::EventHandle<SetUngatedTransferEvent>,
+    }
+
+    /// Struct defined to create a domain separator for wallet object addresses.
+    struct DomainSeparator has drop {}
+
+    /// FungibleAsset can be passed into function for type safety and to guarantee a specific amount.
+    /// FungibleAsset cannot be stored directly and will have to be deposited back into a wallet.
     struct FungibleAsset {
         metadata: Object<FungibleAssetMetadata>,
         amount: u64,
     }
 
-    /// Ref to mint.
+    /// MintRef can be used to mint the fungible asset into an account's wallet.
     struct MintRef has drop, store {
         metadata: Object<FungibleAssetMetadata>
     }
 
-    /// Ref to control the transfer.
+    /// TransferRef can be used to allow or disallow the owner of fungible assets from transferring the asset
+    /// and allow the holder of TransferRef to transfer fungible assets from any account.
     struct TransferRef has drop, store {
         metadata: Object<FungibleAssetMetadata>
     }
 
-    /// Ref to burn..
+    /// BurnRef can be used to burn fungible assets from a given holder account.
     struct BurnRef has drop, store {
         metadata: Object<FungibleAssetMetadata>
     }
 
-    /// The initialization of an object with `FungibleAssetMetadata`.
-    public fun init_metadata(
+    /// Emitted when fungible assets are minted.
+    struct MintEvent has drop, store {
+        amount: u64,
+    }
+
+    /// Emitted when fungible assets are burnt.
+    struct BurnEvent has drop, store {
+        amount: u64,
+    }
+
+    /// Emitted when fungible assets are deposited into a wallet.
+    struct DepositEvent has drop, store {
+        metadata: Object<FungibleAssetMetadata>,
```

Comment:
Yes that's what I was thinking - saving one layer of indirection. It can make the indexing flow a lot simpler. Will check with @bowenyang007 later on this

Reply:
Sure. I'll talk to Bowen later